### PR TITLE
Storage and UserAgent support

### PR DESCRIPTION
**Pull request by talezion**

### CODE REVIEW

File: StanwoodCore/Core/AbstractTableDataSource.swift [hunk 1]
```diff
@@ -183,7 +183,7 @@ extension Stanwood {
                 }
             }
             
-            cell.fill(with: dataType?[indexPath])
+            cell.fill(with: dataType?[indexPath.section][indexPath])
```
Comment (PaddyRocliffe) [inline StanwoodCore/Core/AbstractTableDataSource.swift]: Should this be [indexPath.section][indexPath.row]?

File: StanwoodCore/Core/Storage.swift [hunk 2]
```diff
@@ -49,32 +56,41 @@ extension Stanwood {
...
             switch directory {
-            case .documents:
+            case .documents(customDirectory: let directory):
+                customDirectory = directory
                 searchPathDirectory = .documentDirectory
-            case .caches:
+            case .caches(customDirectory: let directory):
+                customDirectory = directory
                 searchPathDirectory = .cachesDirectory
+            case .library(customDirectory: let directory):
```
Comment (PaddyRocliffe) [inline StanwoodCore/Core/Storage.swift]: Should probably point this at the library subdirectories https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1

File: StanwoodCore/Extensions/UIApplication+Extension.swift [hunk 3]
```diff
@@ -27,10 +27,32 @@ import Foundation
...
+    
+    /**
+     Set the user agent
+     
+     - Parameters:
+        - prefix: Project prefix, i'e gla
+        - env: workspace environment
+     */
+    public func setUserAgent(forProjectPrefix prefix: String, env: Env) {
+        let userAgent = "io-stanwood-\(prefix).\(env.rawValue)-\(UIApplication.shortVersion):\(UIApplication.buildVersion)"
```
Comment (PaddyRocliffe) [inline StanwoodCore/Extensions/UIApplication+Extension.swift]: For public release, we should probably use something like ‘io-\(Organisation)-...’

File: StanwoodCore/Core/Storage.swift (hunk 2 above)
Comment (talezion) [reply StanwoodCore/Core/Storage.swift]: Fixed

File: StanwoodCore/Extensions/UIApplication+Extension.swift (hunk 3 above)
Comment (talezion) [reply StanwoodCore/Extensions/UIApplication+Extension.swift]: I did that wrong actually, we have a different convention for the user agent, fixed

**Pull request merged**